utils/big_decimal.cc

```diff
@@ -143,6 +143,35 @@ std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
     return x.compare(y) <=> 0;
 }
 
+std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
+{
+    if (_scale == other._scale) {
+        return _unscaled_value.compare(other._unscaled_value) <=> 0;
+    }
+
+    const static double sqrt_10 = std::sqrt(10);
+
+    const int64_t bit_count = boost::multiprecision::msb(_unscaled_value) - boost::multiprecision::lsb(_unscaled_value);
+    const int64_t bit_count_other = boost::multiprecision::msb(other._unscaled_value) - boost::multiprecision::lsb(other._unscaled_value);
```

Comment:
What does msb mean on a negative mantissa?

Reply:
Not sure why you care about the lsb. It's basically random.

Reply:
I don't understand this algorithm at all, so you definitely need a unit test to test it.

Reply:
I added detailed comments to the code, as well as extensive tests in v2.

Reply:
(I also dropped the unnecessary use of `lsb()` in v2).